src/provider/model/AppDirectory.ts

```diff
@@ -31,7 +33,7 @@ export class AppDirectory extends AsyncInit {
 
     public async getAppByName(name: AppName): Promise<Application | null> {
         return this._directory.find((app: Application) => {
-            return app.name === name;
+            return (app.name === name || checkCustomConfigField(app, CustomConfigFields.OPENFIN_APP_NAME) === name);
```

Comment:
Do we definitely want to allow a custom field in this case? I can see the logic in the UUID case - a developer for some reason can't have their app UUID match the directory appId, but I don't see what the use case is here.

Reply:
Ok, that was actually my original idea, but then got probably a bit sidetracked by this comment:

```
Understand the changes in sample-app-directory means the new code path does get exercised, but can this be explicitly tested? I think a near-identical test just below the first test in open.inttest.ts, but with an app that uses the fallback customConfig would cover it
```

the first test in open.inttest.ts tests the name field indirectly since `open` makes a call to `getAppByName`

Reply:
I will remove this and add a unit test for `AppDirectory ` to test the `customConfig` entries.